src/Compilers/VisualBasic/Portable/Symbols/Source/SourceMemberContainerTypeSymbol.vb

```diff
@@ -3203,19 +3233,88 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
+                    If tree Is syntaxRef.SyntaxTree Then
+                        Return aggregateLength + position
                     End If
 
                     aggregateLength += syntaxRef.Span.Length
                 Next
+
+                ' This point should not be reachable.
+                Debug.Assert(False)
+                Return -1
+            End If
+
+            Dim syntaxOffset As Integer
+            If TryCalculateSyntaxOffsetOfPositionInInitializer(position, tree, isShared, syntaxOffset:=syntaxOffset) Then
+                Return syntaxOffset
+            End If
+
+            ' This point should not be reachable. An implicit constructor has no body and no initializer,
+            ' so the variable has to be declared in a member initializer.
+            Debug.Assert(False)
+            Return -1
+        End Function
+
+        ' Calculates a syntax offset of a syntax position that is contained in a property or field initializer (if it is in fact contained in one).
+        Friend Function TryCalculateSyntaxOffsetOfPositionInInitializer(position As Integer, tree As SyntaxTree, isShared As Boolean, ByRef syntaxOffset As Integer) As Boolean
+            Dim membersAndInitializers = GetMembersAndInitializers()
+            Dim allInitializers = If(isShared, membersAndInitializers.StaticInitializers, membersAndInitializers.InstanceInitializers)
+
+            Dim siblingInitializers = GetInitializersInSourceTree(tree, allInitializers)
+            Dim index = IndexOfInitializerContainingPosition(siblingInitializers, position)
+            If index < 0 Then
+                syntaxOffset = 0
+                Return False
+            End If
+
+            '                                 |<-----------distanceFromCtorBody---------->|
+            ' [      initializer 0    ][ initializer 1 ][ initializer 2 ][ initializer 3 ][ctor body]
+            ' |<--preceding init len-->|      ^
+            '                              position 
+            Dim initializersLength = If(isShared, membersAndInitializers.StaticInitializersSyntaxLength, membersAndInitializers.InstanceInitializersSyntaxLength)
+            Dim distanceFromInitializerStart = position - siblingInitializers(index).Syntax.Span.Start
+            Dim distanceFromCtorBody = initializersLength - (siblingInitializers(index).PrecedingInitializersLength + distanceFromInitializerStart)
+
+            Debug.Assert(distanceFromCtorBody > 0)
+
+            ' syntax offset 0 is at the start of the ctor body:
+            syntaxOffset = -distanceFromCtorBody
+            Return True
+        End Function
+
+        Private Shared Function GetInitializersInSourceTree(tree As SyntaxTree, initializers As ImmutableArray(Of ImmutableArray(Of FieldOrPropertyInitializer))) As ImmutableArray(Of FieldOrPropertyInitializer)
+            Dim builder = ArrayBuilder(Of FieldOrPropertyInitializer).GetInstance()
+            For Each siblingInitializers In initializers
```

Comment:
Please declare loop variables with an explicit 'As' clause to avoid accidental reuse of an existing field or proeprty.